meilisearch-schema/src/schema.rs

```diff
@@ -0,0 +1,224 @@
+use std::collections::{HashMap, HashSet};
+
+use serde::{Serialize, Deserialize};
+
+use crate::{FieldsMap, FieldId, SResult, Error, IndexedPos};
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct Schema {
+    fields_map: FieldsMap,
+
+    identifier: FieldId,
+    ranked: HashSet<FieldId>,
+    displayed: HashSet<FieldId>,
+
+    indexed: Vec<FieldId>,
+    indexed_map: HashMap<FieldId, IndexedPos>,
+
+    must_index_new_fields: bool,
+}
+
+impl Schema {
```

Comment:
I think we must rework all these methods together, there is to much to write here.

Reply:
I let you review the work done on this file